### PR TITLE
fix: remove internalKeys from Selected row and Triggered row

### LINKED ISSUE #9837: [Bug]: using internal keys in table wiget causes unexpected behaviours
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

unexpected behaviours occurs when using keys like `__primaryKey__`, `__originalIndex__` in table widget

### Steps To Reproduce

1. Add two table widgets to the canvas
2. In `Table1` use the following as tableData 
```js
{{[
  {
    "step": "#1",
    "task": "Drop a table",
    "status": "✅",
    "action": "",
		"selected": [{__primaryKey__: "60000"}, {_id: 6000}, {'__originalIndex__': 23}]
  },
  {
    "step": "#2",
    "task": "Create a query fetch_users with the Mock DB",
    "status": "--",
    "action": "",
		"selected": []
  },
  {
    "step": "#3",
    "task": "Bind the query using => fetch_users.data",
    "status": "--",
    "action": "",
		"selected": [{}]
  },
	{
    "step": "#3",
    "task": "Bind the query using => fetch_users.data",
    "status": "--",
    "action": "",
  }
]}}
```
3. in `Table2` use the following `{{Table1.selectedRow.selected}}`
4. Then select the first row in `Table1`

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

Remove __primaryKey__ and __originalIndex__ from selectedRow, selectedRows and triggeredRow

Fixes #9837

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Jest Test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/remove-internalKeys-from-Selected-row-and-Triggered-row 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.37 **(0.02)** | 37.83 **(0.03)** | 35.97 **(0.01)** | 56.59 **(0.02)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :green_circle: | app/client/src/widgets/TableWidget/widget/derived.js | 69.77 **(4.68)** | 66.27 **(4.73)** | 55.81 **(4.65)** | 68.18 **(5.1)**</details>

### CODE REVIEW

File: app/client/src/widgets/TableWidget/widget/derived.js [hunk 1]
```diff
@@ -25,15 +25,16 @@ export default {
     }
     const filteredTableData =
       props.filteredTableData || props.sanitizedTableData || [];
+    const omitInternalKeys = ["__originalIndex__", "__primaryKey__"];
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidget/widget/derived.js:28]: Lets call this `internalKeysToOmit`

**Review by sbalaji1192 — APPROVED** [review]

**Comment by chandannkumar** [comment]
Tested this PR and working as expected

**Pull request merged**